lib/loaders/file.js

```diff
@@ -26,9 +27,21 @@ module.exports = function (refValue, options) {
 
   let newValue
   try {
-    var data = fs.readFileSync(filePath)
-    newValue = JSON.parse(data)
-  } catch (e) {}
+    var data = fs.readFileSync(filePath, 'utf8')
+    newValue = parseJson(data) || parseYaml(data)
```

Comment:
Is it a good approach to just try parsing both and see which one doesn't fail? Maybe this should be a fallback logic when it is impossible to determine format unambiguously based on filePath extension?

Reply:
Good suggestion, I currently can't test that because currently a ref would only mapped to file in case it has json, yaml or yml extension